component/org.wso2.carbon.identity.oauth2.token.handler.clientauth.mutualtls/src/main/java/org/wso2/carbon/identity/oauth2/token/handler/clientauth/mutualtls/MutualTLSClientAuthenticator.java

```diff
@@ -174,16 +186,18 @@ public boolean authenticateClient(HttpServletRequest request, Map<String, List>
     public boolean canAuthenticate(HttpServletRequest request, Map<String, List> bodyParams,
                                    OAuthClientAuthnContext context) {
 
-        if (clientIdExistsAsParam(bodyParams) && validCertExistsAsAttribute(request)) {
+        String headerName = IdentityUtil.getProperty(MutualTLSUtil.MTLS_AUTH_HEADER);
+        if (clientIdExistsAsParam(bodyParams) && (validCertExistsAsAttribute(request) || (headerName != null &&
+                getCertificateFromHeader(request).isPresent()))) {
```

Comment:
Also it would be better to enable retrieving certificate from header with a config. Otherwise it allows any person to send the public certificate of a client and authenticate easily (since public certificate is accessible to anyone) without using mutual TLS.

Reply:
This config should be disabled by default in the product.

Reply:
Yes. It's disabled by default. The default value was not set to the config. User should add the particular config to enable this.